src/frontend/planner_test/tests/testdata/delta_join.yaml

```diff
@@ -0,0 +1,14 @@
+# This file is automatically generated. See `src/frontend/planner_test/README.md` for more information.
+- sql: |
+    create table a (a1 int, a2 int);
+    create index i_a1 on a(a1);
+    create table b (b1 int, b2 int);
+    create index i_b1 on b(b1);
+    /* should generate delta join plan, and stream index scan */
+    select * from a join b on a.a1 = b.b1 ;
+  stream_plan: |
+    StreamMaterialize { columns: [a1, a2, b1, b2, i_a1.a._row_id(hidden), i_b1.b._row_id(hidden)], pk_columns: [i_a1.a._row_id, i_b1.b._row_id, a1, b1] }
+    └─StreamExchange { dist: HashShard(i_a1.a1, i_b1.b1, i_a1.a._row_id, i_b1.b._row_id) }
+      └─StreamDeltaJoin { type: Inner, predicate: i_a1.a1 = i_b1.b1, output: [i_a1.a1, i_a1.a2, i_b1.b1, i_b1.b2, i_a1.a._row_id, i_b1.b._row_id] }
+        ├─StreamIndexScan { index: "i_a1", columns: [i_a1.a1, i_a1.a2, i_a1.a._row_id], pk: [i_a1.a._row_id], dist: HashShard(i_a1.a1) }
+        └─StreamIndexScan { index: "i_b1", columns: [i_b1.b1, i_b1.b2, i_b1.b._row_id], pk: [i_b1.b._row_id], dist: HashShard(i_b1.b1) }
```

Comment:
Should this be `UpstreamHashShard` physically? 🤔

Reply:
BTW, I've tested creating this materialized view with 2 parallelisms of `i_a1` and 4 of `i_b1`, the meta actor graph builder panicked. 🥵 Seems something wrong.

<img width="1094" alt="image" src="https://user-images.githubusercontent.com/25862682/212840984-c2e645ef-c5de-4887-9443-cf2e3f175e89.png">

Reply:
I think the problem is that we build an actor graph from a fragment graph using topological sort which would violate the `NoShuffle` edge requirement. Reverse topological order seems more reasonable.